ExampleMod/Content/Items/ExampleResearchPresent.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria;
+using Terraria.GameContent.Creative;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Items
+{
+	public class ExampleResearchPresent : ModItem
+	{
+
+		public override void SetStaticDefaults() {
+			DisplayName.SetDefault("Example Present");
+			Tooltip.SetDefault("Contains a random accessory! Try researching it for infinite of it!");
+			CreativeItemSacrificesCatalog.Instance.SacrificeCountNeededByItemId[Type] = ItemLoader.ItemCount;
+		}
+
+		public override void SetDefaults() {
+			Item.CloneDefaults(ItemID.GoodieBag);
+		}
+
+		/*this allows for the present to be researched even when you already have infinite of them*/
+        public override bool? CanResearch() {
+			return true;
+        }
+
+        public override void OnResearched(bool fullyResearched) {
+			/*If you managed to be lucky enough to get all presents, it will give you all
+			 accessories*/
+			if (fullyResearched) {
+				for (int i = 1; i < ItemLoader.ItemCount; i++) {
+					Item tr = new Item(i);
```

Comment:
Use `ContentSamples.ItemsByType` to query item properties, rather than making a new item. And also below on line 47

Reply:
didn't know about that function. Will apply it where it's relevant.